altair/utils/_importers.py

```diff
@@ -95,3 +101,24 @@ def pyarrow_available() -> bool:
         return True
     except (ImportError, RuntimeError):
         return False
+
+
+def import_toolz_function(
```

Comment:
Marked for removal: `altair.utils._importers.import_toolz_function`.

Alternatively, could be made generic, if this functionality is useful for other modules/packages?

Reply:
I don't know if this is useful for any other module, @mattijn do you know?

Reply:
Nope. No idea.

Reply:
Realising now that comment hasn't highlighted the contents of the function. 

Also the deprecation part might be distracting from the part that could be useful.
```py
...
try:
    module = import_module("toolz.curried")
except ImportError as err:
    msg = f"{msg} {err.args[0]}"
    raise ImportError(msg) from err
...
``` 

The other functions in `_importers`:
- `import_pyarrow_interchange`
- `import_vl_convert`
- `import_vegafusion`

They all do fairly similar things, although based on version constraints and use regular imports, like `import vl_convert as vlc` instead of `import_module("vl_convert")`:

<details>
<summary>import_vl_convert</summary>

```py
def import_vl_convert() -> ModuleType:
    min_version = "1.3.0"
    try:
        version = importlib_version("vl-convert-python")
        if Version(version) < Version(min_version):
            raise RuntimeError(
                f"The vl-convert-python package must be version {min_version} or greater. "
                f"Found version {version}"
            )
        import vl_convert as vlc

        return vlc
    except ImportError as err:
        raise ImportError(
            f"The vl-convert Vega-Lite compiler and file export feature requires\n"
            f"version {min_version} or greater of the 'vl-convert-python' package. \n"
            f"This can be installed with pip using:\n"
            f'   pip install "vl-convert-python>={min_version}"\n'
            "or conda:\n"
            f'   conda install -c conda-forge "vl-convert-python>={min_version}"\n\n'
            f"ImportError: {err.args[0]}"
        ) from err
```

</details>


So rather than having a **dedicated function per import** to maintain and keep consistent - you could do something like below. 
I've taken this from my own code I was working on during #3420, as it shows a larger number of optional dependencies: 

```py
from types import ModuleType
from typing import NamedTuple

class ModuleAvailable(NamedTuple):
    module: ModuleType | None = None
    available: bool = True


_init_modules = (
    "msgspec",
    "tomli_w",
    "aiofiles",
    "xlsxwriter",
    "plotly",
    "altair",
    "lxml",
    "ftfy",
    "playwright",
    "keyring",
)
_available_modules: dict[str, ModuleAvailable] = {
    name: ModuleAvailable() for name in _init_modules
}

def import_optional(
    module_name: str,
    /,
    scope: _UserDefinedScope | str | Literal["ignore"],
    extra: str = "",
) -> ModuleType:
    if item := _available_modules.get(module_name):
        if module := item.module:
            return module
        elif not item.available:
            msg = _optional_module_msg(scope, module_name, extra) # defined elsewhere
            raise ModuleNotFoundError(msg)
    return _import(module_name, scope, extra)

def _import(
    module_name: str, /, scope: _UserDefinedScope | str, extra: str
) -> ModuleType:
    global _available_modules
    lib_name = module_name.split(".", 1)[0]
    try:
        module = import_module(module_name)
        _available_modules[module_name] = ModuleAvailable(module=module)
        if module_name != lib_name:
            _available_modules[lib_name] = ModuleAvailable(module=sys.modules[lib_name])
    except ImportError:
        unavailable = ModuleAvailable(available=False)
        _available_modules[module_name] = unavailable
        if module_name != lib_name:
            _available_modules[lib_name] = unavailable
        msg = _optional_module_msg(scope, module_name, extra) # defined elsewhere
        raise ModuleNotFoundError(msg) from None
    return module
```

Something like [polars.dependencies](https://github.com/pola-rs/polars/blob/687dee3095ae00137ff7fdb31533ecf960b324e7/py-polars/polars/dependencies.py) is a much more developed spin on this - particularly [dependencies.import_optional](https://github.com/pola-rs/polars/blob/687dee3095ae00137ff7fdb31533ecf960b324e7/py-polars/polars/dependencies.py#L231) which has version bounds.

Reply:
Just to be clear, this is more me floating an idea - not something that needs to be tied to this PR

Reply:
Thanks for mentioning this suggestions! Can we remove this `import_toolz_function()` function? I don't think it is used anywhere anymore.

Reply:
Sure thing @mattijn

Fixed in 88fbd1141b39ca74b47106d254969661e77bdabf